### PR TITLE
Fix MLflow registry methods with empty metadata

### PR DESCRIPTION
**Pull request by fa9r**
## Describe changes
The following model registry methods were all broken if no metadata was provided:
- register_model_version
- update_model_version
- list_model_versions

Problem: the function declarations specified `pydantic.Field` as default value which is only viable for Pydantic models and not for regular Python functions.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: examples/e2e/steps/inference/inference_get_current_version.py [hunk 1]
```diff
@@ -40,7 +40,6 @@ def inference_get_current_version() -> Annotated[str, "model_version"]:
 
     current_version = model_registry.list_model_versions(
         name=MetaConfig.mlflow_model_name,
-        metadata={},
```
Comment (avishniakov) [inline examples/e2e/steps/inference/inference_get_current_version.py:43]: This should be fixed on template side - there is a CI auto-refreshing it.

File: examples/e2e/steps/promotion/promote_get_versions.py [hunk 2]
```diff
@@ -48,15 +48,13 @@ def promote_get_versions() -> (
     ### ADD YOUR OWN CODE HERE - THIS IS JUST AN EXAMPLE ###
     none_versions = model_registry.list_model_versions(
         name=MetaConfig.mlflow_model_name,
-        metadata={},
```
Comment (avishniakov) [inline examples/e2e/steps/promotion/promote_get_versions.py:51]: This should be fixed on template side - there is a CI auto-refreshing it.

File: examples/e2e/steps/promotion/promote_get_versions.py [hunk 3]
```diff
@@ -48,15 +48,13 @@ def promote_get_versions() -> (
...
         stage=None,
     )
     latest_versions = none_versions[0].version
     logger.info(f"Latest model version is {latest_versions}")
 
     target_versions = model_registry.list_model_versions(
         name=MetaConfig.mlflow_model_name,
-        metadata={},
```
Comment (avishniakov) [inline examples/e2e/steps/promotion/promote_get_versions.py:59]: This should be fixed on template side - there is a CI auto-refreshing it.

File: examples/e2e/steps/promotion/promote_metric_compare_promoter.py [hunk 4]
```diff
@@ -88,13 +88,11 @@ def promote_metric_compare_promoter(
                 name=MetaConfig.mlflow_model_name,
                 version=current_version,
                 stage=ModelVersionStage.ARCHIVED,
-                metadata={},
```
Comment (avishniakov) [inline examples/e2e/steps/promotion/promote_metric_compare_promoter.py:91]: This should be fixed on template side - there is a CI auto-refreshing it.

File: examples/e2e/steps/promotion/promote_metric_compare_promoter.py [hunk 5]
```diff
@@ -88,13 +88,11 @@ def promote_metric_compare_promoter(
...
             )
         model_registry.update_model_version(
             name=MetaConfig.mlflow_model_name,
             version=latest_version,
             stage=MetaConfig.target_env,
-            metadata={},
```
Comment (avishniakov) [inline examples/e2e/steps/promotion/promote_metric_compare_promoter.py:97]: This should be fixed on template side - there is a CI auto-refreshing it.

**Review by avishniakov — CHANGES_REQUESTED** [review]
The template needs a fix in another place, rest is good to go.

File: examples/e2e/steps/inference/inference_get_current_version.py (hunk 1 above)
Comment (avishniakov) [reply examples/e2e/steps/inference/inference_get_current_version.py:43]: https://github.com/zenml-io/template-e2e-batch/pull/10

**Review by avishniakov — APPROVED** [review]
After CI completes and brings template updates here - good to go 👍🏼

**Review by strickvl — APPROVED** [review]

**Pull request merged**